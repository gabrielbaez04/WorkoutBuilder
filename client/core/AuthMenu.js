import React from 'react'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import {Link, withRouter} from 'react-router-dom'
import auth from './../auth/auth-helper'
import IconButton from '@material-ui/core/IconButton';
import {withStyles} from '@material-ui/core/styles'

const isActive = (history, path) => {
    if (history.location.pathname == path)
      return {color: '#ff4081'}
    else
      return {color: '#ffffff'}
  }
  
class AuthMenu extends React.Component{
    state = {
        class: 'topnav',
    };
    handleMenuClick = () => {
        this.setState({ class: this.state.class == 'topnav' ? 'topnav responsive' : 'topnav'});
    };

    update = (props) =>{
        auth.signout(() => this.props.history.push('/'));
        this.props.updateMenu();
        
    }
    

    render(){
        const css = require('./Menu.css');
        const { history, classes } = this.props;
        return(
            <div className={this.state.class} id="myTopnav">
                <Link to={"/user/" + this.props.authenticated.user._id} 
                    style={isActive(history, "/user/" + this.props.authenticated.user._id)} 
                    className="active"
                    onClick={this.handleMenuClick}> My Profile </Link>

                <Link to="/workout" style={isActive(history, "/workout")}
                    onClick={this.handleMenuClick}> Workouts </Link>

                <a onClick={this.update}> Sign out </a>
                
                <a  className="icon" onClick={this.handleMenuClick}>
                    <i className="fa fa-bars"></i>
                </a>
            </div>
      )
    }

}
export default withRouter(AuthMenu);