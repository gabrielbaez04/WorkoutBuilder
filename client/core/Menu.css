.topnav {
    overflow: hidden;
    background-color: #333;
    height: 100%;
    z-index: 1;
  }
  
  .topnav a {
    float: left;
    display: block;
    color: #f2f2f2;
    text-align: center;
    padding: 12px 16px;
    text-decoration: none;
    font-size: 17px;
    
  }
  
  .topnav a:hover {
    background-color: #6b6969;
  }
  
  .topnav .icon {
    display: none;
  }
  
  @media screen and (max-width: 600px) {
    .topnav a{display: none;}
    .topnav a:not(:last-child){padding-right:55px;}
    .topnav a.icon {
      float: right;
      display: block;

    }
  }

  @media screen and (max-width: 600px) {
    .topnav {
        position: absolute;
        right: 0;
        top: 0;
      }
      .topnav.responsive{
        height: auto;
      }
      .topnav .icon {
        height: 100%;
      }
    .topnav.responsive .icon {
      position: absolute;
      right: 0;
      top: 0;
      height: auto;
    }
    .topnav.responsive a {
      float: none;
      display: block;
      text-align: left;
    }
  }