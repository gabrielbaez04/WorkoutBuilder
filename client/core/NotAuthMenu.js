import React from 'react'
import Typography from 'material-ui/Typography'
import Button from 'material-ui/Button'
import {Link, withRouter} from 'react-router-dom'

const NotAuthMenu = withRouter((props) =>(
        <span>
            <Link to="/signup">
                <Button style={props.style(props.history, "/signup")}>Sign up
                </Button>
            </Link>
            <Link to={{pathname: "/signin", updateMenu: props.updateMenu}}>
                <Button style={props.style(props.history, "/signin")}>Sign In
                </Button>
            </Link>
        </span>
    ));

export default NotAuthMenu;