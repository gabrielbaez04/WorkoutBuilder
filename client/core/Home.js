import React from 'react'
import {withStyles} from '@material-ui/core/styles'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography'
import seashellImg from './../assets/images/seashell.jpg'

const styles = theme => ({
  card: {
    maxWidth: '90%',
    margin: 'auto',
    marginTop: theme.spacing.unit * 5
  },
  title: {
    padding:`${theme.spacing.unit * 3}px ${theme.spacing.unit * 2.5}px ${theme.spacing.unit * 2}px`,
    color: theme.palette.text.secondary
  },
  media: {
    minHeight: 330
  }
})

const Home = (props)=> {
    const {classes} = props
    return (
        <Card className={classes.card}>
          <Typography variant="display1" className={classes.title}>
            Home Page
          </Typography>
          <CardMedia className={classes.media} image={seashellImg} title="Unicorn Shells"/>
          <CardContent>
            <Typography>
              Welcome to the MERN Skeleton home page.
            </Typography>
          </CardContent>
        </Card>
    )
}


export default withStyles(styles)(Home)
