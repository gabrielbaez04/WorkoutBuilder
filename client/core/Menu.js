import React from 'react'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import HomeIcon from '@material-ui/icons/Home'
import Button from '@material-ui/core/Button'
import auth from './../auth/auth-helper'
import {Link, withRouter} from 'react-router-dom'
import AuthMenu from './AuthMenu';
import NotAuthMenu from './NotAuthMenu';

class Menu extends React.Component{
  state = {
    auth: false,
  }
  componentDidMount = () =>{
    this.setState({auth:auth.isAuthenticated()})
  }
  handleUpdate = () =>{
    this.setState({auth:auth.isAuthenticated()})
  }
  render()
  {
    const isActive = (history, path) => {
      if (this.props.history.location.pathname == path)
        return {color: '#ff4081'}
      else
        return {color: '#ffffff'}
    }
    return(
      <AppBar position="static">
        <Toolbar>
        <Link to="/">
          <Typography variant="display1" color="secondary" aria-label="Home" style={isActive(this.props.history, "/")}>
            Workout Builder
          </Typography>
        </Link>
          {
            this.state.auth 
            ? <AuthMenu style={isActive} authenticated={this.state.auth} updateMenu={this.handleUpdate}/> 
            : <NotAuthMenu style={isActive} updateMenu={this.handleUpdate}/>
          }
        </Toolbar>
      </AppBar>
      )
    }
 
}

export default withRouter(Menu);