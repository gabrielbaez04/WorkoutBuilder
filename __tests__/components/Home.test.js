import { Home } from '../../client/core/Home'
import React from 'react';
import { shallow } from 'enzyme';
import advices from '../fixtures/advices';

test('should render Home correctly with 3 advices',()=>{
  const wrapper = shallow(<Home classes={{}}/>);
  wrapper.setState({cards: advices})
  expect(wrapper).toMatchSnapshot();
});

test('should render Home correctly with 3 advices',()=>{
  const wrapper = shallow(<Home classes={{}}/>);
  expect(wrapper.find('WithStyles(Card)').length).toBe(3);
});
