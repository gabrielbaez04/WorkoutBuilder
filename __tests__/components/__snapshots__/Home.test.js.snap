// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home correctly with 3 advices 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <div>
    <main>
      <div
        className=""
      >
        <WithStyles(Grid)
          container={true}
          spacing={40}
        >
          <WithStyles(Grid)
            item={true}
            key="0"
            lg={3}
            md={4}
            sm={6}
          >
            <WithStyles(Card)>
              <WithStyles(CardMedia)
                image="test-file-stub"
                title="Image title"
              />
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                  component="h2"
                  gutterBottom={true}
                  variant="headline"
                >
                  Make sure you're eating healthy.
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  variant="subheading"
                >
                  Ask almost any personal trainer and they’ll tell you that regardless of your training goals, healthy eating is the backbone. Food is what fuels your body to reach your goals, and without proper nutrition through quality foods, you’re likely to stall. Maintain a balanced diet consisting of fruits, vegetables, complex carbohydrates, complete proteins, and healthy fats like fish oils and flaxseeds.
                </WithStyles(Typography)>
              </WithStyles(CardContent)>
            </WithStyles(Card)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            key="1"
            lg={3}
            md={4}
            sm={6}
          >
            <WithStyles(Card)>
              <WithStyles(CardMedia)
                image="test-file-stub"
                title="Image title"
              />
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                  component="h2"
                  gutterBottom={true}
                  variant="headline"
                >
                  Prepare ahead.
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  variant="subheading"
                >
                  Preparing meals in advance gives you the best chance to accomplish your nutrition goals, says Micah LaCerte, a personal trainer and fitness competition world champion. That way, he says, you won’t feel pressured to eat unhealthy foods or skip meals.
                </WithStyles(Typography)>
              </WithStyles(CardContent)>
            </WithStyles(Card)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            key="2"
            lg={3}
            md={4}
            sm={6}
          >
            <WithStyles(Card)>
              <WithStyles(CardMedia)
                image="test-file-stub"
                title="Image title"
              />
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                  component="h2"
                  gutterBottom={true}
                  variant="headline"
                >
                  Eat more clean food.
                </WithStyles(Typography)>
                <WithStyles(Typography)
                  variant="subheading"
                >
                  Eating only three daily meals? Not a great idea. “Half the people I deal with aren’t losing weight because they don’t eat enough,” says veteran personal trainer Mike Duffy. Duffy advises his clients “to eat five times a day, about every three hours, to stimulate their metabolism” including two mini-meals between three basic meals. With activity levels decreasing throughout the day, he advises to “eat less as the day goes on.”
                </WithStyles(Typography)>
              </WithStyles(CardContent)>
            </WithStyles(Card)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </div>
    </main>
    <footer>
      <WithStyles(Typography)
        align="center"
        color="textSecondary"
        component="p"
        variant="subheading"
      >
        Developed and designed by 
        <a
          href="https://github.com/gabrielbaez04"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gabriel Baez
        </a>
      </WithStyles(Typography)>
    </footer>
  </div>
</Fragment>
`;
